mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -668,6 +668,12 @@ def WMMAInTypes : AnyTypeOf<[VectorOfLengthAndType<
 def WMMAOutTypes : AnyTypeOf<[VectorOfLengthAndType<[4, 8], [F32, I32]>,
                               VectorOfLengthAndType<[4, 8, 16], [F16, BF16]>]>;
 
+// MI300: limit to only 4 bytes per elements.
+def GlobalLoadTypes : AnyTypeOf<[F16, F32, I8, SI8, UI8, I16, I32,
```

Comment:
Let's not do that

This works for 1, 2, or 4 bytes, and can be a check in `::verify`

Reply:
done. I changed to restrict the type sizes in `::verify`.